internal/invocation/invocation_service.go

```diff
@@ -183,7 +190,16 @@ func (s *Service) handleClientMessage(msg *proto.ClientMessage) {
 				return fmt.Sprintf("invocation with unknown correlation ID: %d", correlationID)
 			})
 		} else if inv.EventHandler() != nil {
-			go inv.EventHandler()(msg)
+			handler := func() {
```

Comment:
How about making `executor.dispatch` and `executor.dispatchRandom` take a `proto.ClientMessageHandler` insted of `func()` so this extra handler is not required?